src/coreclr/jit/emitarm64.cpp

```diff
@@ -6932,8 +7001,18 @@ void emitter::emitIns_R_R_R_I(instruction ins,
         }
     }
 
-    dispIns(id);
-    appendToCurIG(id);
+    // Now the instruction is EITHER emitted OR used to overwrite the previously-emitted instruction.
+    if (reuseInstr == nullptr)
+    {
+        // Then this is the standard exit path and the instruction is to be appended to the instruction group.
+        dispIns(id);
+        appendToCurIG(id);
+    }
+    else
+    {
+        // The instruction is copied over the last emitted insdtruction.
+        memcpy(reuseInstr, id, sizeof(tempInstrDesc));
```

Comment:
I assume it is guaranteed that `reuseInstr` and `id` have the same size?
It would be good to add `assert(emitGetInstrDescSize(reuseInstr) == emitGetInstrDescSize(id))` before this.